packages/track-event-mixin/track-event-mixin.ts

```diff
@@ -0,0 +1,46 @@
+import { LitElement, PropertyValues } from 'lit-element';
+import { Constructor, applyMixin, wasApplied } from '@vaadin/mixin-utils';
+import { TrackEventClass } from './track-event-class';
+import { addTrackListeners, removeTrackListeners } from './track-event-gesture';
+
+export const TrackEventMixin = <T extends Constructor<LitElement>>(
+  base: T
+): T & Constructor<TrackEventClass> => {
+  if (wasApplied(TrackEventMixin, base)) {
+    return base;
+  }
+
+  class TrackEvent extends base {
+    protected firstUpdated(props: PropertyValues) {
+      super.firstUpdated(props);
+
+      if (Array.isArray(this._trackTargets)) {
+        this._trackTargets.forEach((target: Element) => {
+          if (target instanceof HTMLElement) {
+            addTrackListeners(target);
+          }
+        });
+      }
+    }
+
+    protected get _trackTargets() {
+      return [this];
+    }
+
+    disconnectedCallback() {
```

Comment:
If the element is, for example, relocated, this would remove its track listeners, right? But there doesn't seem to be a `connectedCallback`, that would add the listeners back. `firstUpdated` adds them, but it's only run once during the elements lifecycle.

Reply:
Thanks, changed to use `connectedCallback` and then wait for `updateComplete`.